CHERI-UPDATING.md

```diff
@@ -79,6 +79,156 @@ Each merge from upstream FreeBSD to [dev] is accompanied by a tag of the form
+sub-optimal) proxy for the time the change was pushed to the FreeBSD
+tree. We currently do this by examining the log and looking for the
+first `CommitDate: ` line mentioning the day we wish to merge. To see
+only eligible commits and to cause the commit date to be displayed we
+use:
+```
+$ git log --first-parent --format=fuller upstream/main
+```
+We can then use the pager's search function to search for the
+appropriate date (e.g.. `Commit.*Fri Feb 12`). It's a good idea to check
+a few commits up or down to make sure the commit isn't at an obviously
+bad point. Then we create a tag for the commit:
+```
+$ git tag -m "FreeBSD main DD Month YYYY" freebsd-main-<YYYYMMDD> <hash>
+```
+
+#### Performing the merge
+First, create a merge branch from an update to date `dev` checkout:
+```
+$ git checkout -b merge-freebsd-<YYYYMMDD> dev
+```
+Then start the merge:
+```
+$ mergify start freebsd-main-<YYYYMMDD>
+```
+The mergify command will start merging each change one at a time until
+it hits a conflict or a pause point (e.g., a path in the merged commit
+matches the regex in [.mergify_pause_paths].) After inspecting the
+change or resolving the conflict, continue with:
+```
+mergify continue
+```
+
+For ease of conflict resolution, `mergify resolve` opens each file
+containing conflict files in `EDITOR` in turn. `mergify autoadd` adds
+all conflicted files that no longer contain conflict markers to the
+index. It does not add files that were not in the original merge, they
+need to be added with `git add`.
+
+Sometimes a change has already been applied to the tree or is a change
+requiring porting that is reverted later on. In that case, `mergify
+skip` can be used. It restores all modified files to the version in HEAD
+and then continues.
+
+#### Rolling back
+Over the course of the merge process, it is not uncommon to discover
+that a previously merged change requires additional changes. Ideally
+we'd make a fixup commit and then rebase, but `git rebase` is useless
+with merge commits. Instead, the general process is to stop the merge,
+use `git reset` to rewind to the broken commit, fix it up, and restart
+the merge. The process usually looks like
+```
+$ mergify abort
+$ git reset --hard <bad merge>
+...
+$ git commit --amend
+$ mergify start freebsd-main-<YYYYMMDD>
+```
+Sometime the commit itself isn't the problem, but rather that it
+enabled a feature than was broken with CHERI. When this happens,
+resetting to the commit before the feature was enabled, fixing things,
+and restarting can make sense.
+
+Sometimes there are too many awkward changes (e.g. one that required
+fixup in files outside the original change) and it's not worth
+backtracking to fix things in the ideal location. In that case, make
+sure to indicate that the fixup commit fixes an issue introduced in a
+particular commit to aid future users of bisect.
+
+#### Finishing up
+Once the merge is complete and basic local testing is done, it's time to
```

Comment:
One note I guess is that I've been deferring doing these steps until the merge PR completes successfully and push all three things (new tag, freebsd-main, dev) at once.